### PR TITLE
feat(Responsive): add getWidth prop, fix perf

### PR DESCRIPTION
**Pull request by andrewferk**
Implements #2614 

1. Revert #2531, as it breaks `Responsive`'s integration with Semantic UI's breakpoints. This ensures `Responsive` uses `window.innerWidth` as the default calculation for screen width. `document.documentElement.clientWidth` does not use the scrollbar in its calculation.

2. Fix false-positive tests in `Responsive-test.js`. [`chai-enzyme`](https://github.com/producthunt/chai-enzyme) provides assertion helpers [`blank()`](https://github.com/producthunt/chai-enzyme#blank) and [`present()`](https://github.com/producthunt/chai-enzyme#present), which were being used as if they were opposites. They are not opposites. For a wrapper to be `empty()`, it must also be `present()`. Tests have been updated to use `not.empty()` and `empty()`, which are opposites. Also, the resizing test was using enzyme's `shallow()`, but `mount()` was needed to support state change and updating render.

3. Add optional `getWidth` property to `Responsive`. The property is a function that returns the width of the screen. This property defaults to a function that returns `window.innerWidth` for browsers and `0` for non-browsers. @Autarc or anyone else wanting to use `document.documentElement.clientWidth` as discussed in #2531 can do so using `<Responsive getWidth={() => document.documentElement.clientWidth} />`.

4. Prevent `Responsive` from updating render when `window.resize` is triggered, but width does not change. `Responsive` currently updates on mobile devices when scrolling and the url bar displays and hides. This happens because the height of the window is changing, triggering `window.resize`. This update allows `Responsive` to ignore these events. `onUpdate` will still be invoked.

### CODE REVIEW

**Comment by andrewferk** [comment]
Maybe the way I did the default `getWidth` doesn't look the best in the docs.

image

I'd be more than welcome to remove it from the `defaultProps` and implement it another way.

**Review by layershifter — APPROVED** [review]
I've added some styling fixed. LGTM

@andrewferk thanks for the awesome work there :+1:

File: src/addons/Responsive/Responsive.js [hunk 1]
```diff
@@ -46,6 +53,10 @@ export default class Responsive extends Component {
     onUpdate: PropTypes.func,
   }
 
+  static defaultProps = {
+    getWidth: () => (isBrowser() ? window.innerWidth : 0),
```
Comment (levithomason) [inline src/addons/Responsive/Responsive.js:57]: I would think if there's no browser at all, `null` might be more appropriate.  Thoughts?

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.80.0`.

**Pull request merged**